### DISCUSSION TITLE
Setting to produce server IDs, user IDs & message links for reporting harmful users & servers?

### DISCUSSION THREAD

**Discussion by TMatt92** [Tyrrrz/DiscordChatExporter#674] [General]
Like the title says.

**Comment by 96-LB** [comment]
(Note: I was already drafting this when you posted that you solved the answer, but I decided to finish it for the sake of completeness in case anyone else runs into this.)

User and message IDs are already accessible within the export. You'll have to open Inspect Element or view the source to get it from the HTML format:
![image](https://user-images.githubusercontent.com/21089343/128458074-04de1602-a235-4c65-8b1d-236608bd9c53.png)

I don't think the server and channel IDs are visible within an HTML export, but you can use [filename templates](https://github.com/Tyrrrz/DiscordChatExporter/wiki/GUI%2C-CLI-and-Formats-explained#generating-the-filename-and-output-directory-dynamically) to expose them in the filename. Alternatively, you can utilize the JSON format instead.

Message links are just the link to the channel with a message ID appended, so you can build them using the format `https://discord.com/​<server_id>/<channel_id>/<message_id>`.
